lib/handler/mruby/middleware.c

```diff
@@ -197,11 +197,15 @@ static mrb_value build_app_response(struct st_mruby_subreq_t *subreq)
     return resp;
 }
 
-static void append_bufs(struct st_mruby_subreq_t *subreq, h2o_iovec_t *inbufs, size_t inbufcnt)
+static void append_bufs(struct st_mruby_subreq_t *subreq, h2o_sendvec_t *inbufs, size_t inbufcnt)
 {
-    int i;
+    size_t i;
     for (i = 0; i != inbufcnt; ++i) {
-        h2o_buffer_append(&subreq->buf, inbufs[i].base, inbufs[i].len);
+        char *dst = h2o_buffer_reserve(&subreq->buf, inbufs[i].len).base;
+        assert(dst != NULL && "no memory or disk space; FIXME bail out gracefully");
+        if (!(*inbufs[i].fill_cb)(&subreq->super, h2o_iovec_init(dst, inbufs[i].len), inbufs + i, 0))
+            h2o_fatal("FIXME handle error from pull handler");
+        subreq->buf->size += inbufs[i].len;
```

Comment:
@i110 Do you have an idea on how we should propagate errors that occur in this block?

The error condition happens when i) `h2o_buffer_reserve` fails to allocate space (happens in master as well), or ii) `fill_cb` returns an error (unique to this PR).